### PR TITLE
Add method to explain query vector

### PR DESCRIPTION
**Pull request by PradyumnaKrishna**
Adds ExplainOutputNode struct with OperatorName and Children, A method `explainVector` generates `ExplainOutputNode` for the provided `VectorOperator`.

This `explainVector` method is used by `Explain` method of query to get an explaination of the query.

### CODE REVIEW

File: engine/engine.go [hunk 1]
```diff
@@ -346,15 +351,29 @@ type Query struct {
...
 }
 
 func (q *Query) Profile() {
 	// TODO(bwplotka): Return profile.
 }
 
+func explainVector(v model.VectorOperator) *ExplainOutputNode {
+	name, vectors := v.Explain()
+
+	var childs []ExplainOutputNode
```
Comment (fpetkovski) [inline engine/engine.go:366]:
```suggestion
	var children []ExplainOutputNode
```

File: engine/engine.go [hunk 2]
```diff
@@ -346,15 +351,29 @@ type Query struct {
...
+	for _, vector := range vectors {
+		childs = append(childs, *explainVector(vector))
```
Comment (fpetkovski) [inline engine/engine.go:368]: Can `explainVector` return `nil`?

**Review by GiedriusS — COMMENTED** [review]
Maybe we could add some tests for this e.g. `Explain()` on a query that hasn't been executed yet, `Explain()` on a query that has one operator, `Explain()` on a "regular" query like `sum(up) by (job)`.

**Comment by PradyumnaKrishna** [comment]
I guess the query `Explain()` method produce output according to no of processors available (`concurrencyOperator `). That's the cause of failing test.

**Comment by GiedriusS** [comment]
> I guess the query `Explain()` method produce output according to no of processors available (`concurrencyOperator `). That's the cause of failing test.

I think we could use `runtime.GOMAXPROCS(0) / 2` to calculate the number of concurrency operators.

**Review by GiedriusS — APPROVED** [review]

**Pull request merged**